### PR TITLE
Nuke the privacy poll, offer ghosts control option, memo update, SQL update, fixes

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Tactically nukes the privacy poll from orbit. Based on this PR by MrStonedOne: https://github.com/tgstation/-tg-station/pull/10440.
2) Adds a new option for admins under View Variables to offer control of a mob to ghosts. By KorPhaeron: https://github.com/tgstation/-tg-station/pull/11896.
3) Makes memos SQL based. By Jordie0608: https://github.com/tgstation/-tg-station/pull/10694.
4) Fixes IPC posibrains not having names while in the mob. Fixes https://github.com/ParadiseSS13/Paradise/issues/2126.
5) Completes my previous work with standardizing SQL, by ditching the "erro_" prefix completely and getting rid of useless database settings. Should prevent new hosts from crashing when they try launching as well.
6) Adds logging to the  "Free Job Slot" verb.
7) Adds a config option to disable lobby music.
8) Instrument tweaks: it's now possible to mute instrument sounds, and the BPM of songs will no longer be affected by lag. By MrPerson: https://github.com/tgstation/-tg-station/pull/8456.
9) Vox/Plasmamen wizards will now spawn with their necessary equipment. Also got rid of an obsolete/snowflakey proc and moved racial equipment to the species equip() proc. Fixes https://github.com/ParadiseSS13/Paradise/issues/1794.
10) Admins with R_PERMISSION can now create server polls (without having to use the database). Also fixes multiple choice polls not working properly. By Jordie0608: https://github.com/tgstation/-tg-station/pull/11919.
11) Vox will now spawn with their equipment during highlander/dodgeball, and plasmamen will be made humans.
12) Everything in the traitor panel is now logged. Also updates the edit memory functionality so it doesn't break HTML.
13) Improves the formatting of admin announcements. By GinjaNinja32: https://github.com/Baystation12/Baystation12/pull/11026.
14) Removes the obsolete edit appearance proc.
15) Adds right checks/logging to various admin tools.
16) Cleans up a lot of rights checks, and ensures admins R_MOD checks also check for R_ADMIN so admins aren't dependent on R_MODERATOR.
17) Fixes empty steal objectives. Fixes https://github.com/ParadiseSS13/Paradise/issues/1815.

SQL changes needed:
```
DROP TABLE `privacy`;
```
And:
```
CREATE TABLE `memo` (
  `ckey` varchar(32) NOT NULL,
  `memotext` text NOT NULL,
  `timestamp` datetime NOT NULL,
  `last_editor` varchar(32),
  `edits` text,
  PRIMARY KEY (`ckey`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
And:
```
ALTER TABLE erro_admin RENAME TO admin;
ALTER TABLE erro_admin_log RENAME TO admin_log;
ALTER TABLE erro_ban RENAME TO ban;
ALTER TABLE erro_feedback RENAME TO feedback;
ALTER TABLE population RENAME TO legacy_population;
ALTER TABLE erro_notes RENAME TO notes;
ALTER TABLE erro_player RENAME TO player;
ALTER TABLE erro_poll_option RENAME TO poll_option;
ALTER TABLE erro_poll_question RENAME TO poll_question;
ALTER TABLE erro_poll_textreply RENAME TO poll_textreply;
ALTER TABLE erro_poll_vote RENAME TO poll_vote;
ALTER TABLE erro_watch RENAME TO watch;
```
And:
```
ALTER TABLE `poll_question` ADD COLUMN `multiplechoiceoptions` INT(2) DEFAULT NULL AFTER `adminonly`, ADD COLUMN `createdby_ckey` VARCHAR(45) NULL DEFAULT NULL AFTER `multiplechoiceoptions`, ADD COLUMN `createdby_ip` VARCHAR(45) NULL DEFAULT NULL AFTER `createdby_ckey`
```
Additionally, the forumdbconfig.txt file can be deleted, and the dbconfig.txt file needs to be updated to this format: https://github.com/Markolie/Paradise/blob/sql/config/example/dbconfig.txt.

Also, the MEDIA_BASE_URL will now need to be defined in config.txt if you use jukeboxes.

### CODE REVIEW

**Comment by tigercat2000** [comment]
:+1: x :100:

File: code/defines/procs/statistics.dm [hunk 1]
```diff
@@ -70,25 +69,23 @@ proc/sql_report_death(var/mob/living/carbon/human/H)
...
 		return
 
-	var/turf/T = get_turf(H)
-	var/area/placeofdeath = get_area(T)
-	var/podname = sanitizeSQL("Unknown Area")
-	if(placeofdeath)
-		podname = sanitizeSQL(placeofdeath.name)
+	var/turf/T = H.loc
+	var/area/placeofdeath = get_area(T.loc)
+	var/podname = placeofdeath.name
```
Comment (tigercat2000) [inline code/defines/procs/statistics.dm]: this probably should `sanitizeSQL()` still; Players can create and rename areas.

File: config/example/dbconfig.txt [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
+## MySQL server port (default is 3306).
 PORT 3306
 
-# Database the population, death, karma, etc. tables may be found in
-DATABASE paradise
-
-# Username/Login used to access the database
-LOGIN mylogin
+## Database for all SQL functions, not just feedback.
+FEEDBACK_DATABASE feedback
```
Comment (tigercat2000) [inline config/example/dbconfig.txt]: this should probably be paradise by default

Comment (Markolie) [inline config/example/dbconfig.txt]: I did update paradise_schema.sql to use feedback as well - it's what /tg/ uses, so I think it's best to have parity there.

File: code/defines/procs/statistics.dm [hunk 3]
```diff
@@ -163,4 +159,5 @@ proc/sql_commit_feedback()
 			var/DBQuery/query = dbcon.NewQuery("INSERT INTO [format_table_name("feedback")] (id, roundid, time, variable, value) VALUES (null, [newroundid], Now(), '[variable]', '[value]')")
 			if(!query.Execute())
 				var/err = query.ErrorMsg()
-				log_game("SQL ERROR during death reporting. Error : \[[err]\]\n")
\ No newline at end of file
+				log_game("SQL ERROR during feedback reporting. Error : \[[err]\]\n")
+
```
Comment (tigercat2000) [inline code/defines/procs/statistics.dm]: end of file with tabs
should just delete the tabs or make it end on the `log_game` line

File: code/datums/datumvars.dm [hunk 4]
```diff
@@ -591,9 +592,37 @@ client
...
+		log_admin("[key_name(usr)] has offered control of ([key_name(M)]) to ghosts.")
+		message_admins("[key_name_admin(usr)] has offered control of ([key_name_admin(M)]) to ghosts")
+		var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as [M.real_name]?", "Player Control", null, FALSE, 100)
+		var/mob/dead/observer/theghost = null
+
+		if(candidates.len)
+			theghost = pick(candidates)
+			M << "Your mob has been taken over by a ghost!"
+			message_admins("[key_name_admin(theghost)] has taken control of ([key_name_admin(M)])")
+			M.ghostize()
```
Comment (tigercat2000) [inline code/datums/datumvars.dm]: ghostize() always makes a dead mob, client or not, if I recall correctly, and pointless dead mobs are indeed pointless

File: code/datums/datumvars.dm [hunk 5]
```diff
@@ -591,9 +592,37 @@ client
...
 		href_list["datumrefresh"] = href_list["make_skeleton"]
+			
+	else if(href_list["offer_control"])
+		if(!check_rights(R_ADMIN))	return
+
+		var/mob/M = locate(href_list["offer_control"])
+		if(!istype(M))
+			usr << "This can only be used on instances of type /mob"
+			return
+		M << "Control of your mob has been offered to dead players."
```
Comment (tigercat2000) [inline code/datums/datumvars.dm]: This is eh. It The player isn't warned for any of the other VV options.

File: code/controllers/configuration.dm [hunk 6]
```diff
@@ -78,10 +78,10 @@
 
 	var/server
 	var/banappeals
-	var/wikiurl = "http://baystation12.net/wiki"
-	var/forumurl = "http://baystation12.net/forums/"
+	var/wikiurl = "http://example.org"
+	var/forumurl = "http://example.org"
 
-	var/media_base_url = "http://nanotrasen.se/media" // http://ss13.nexisonline.net/media
+	var/media_base_url = "http://example.org"
```
Comment (tigercat2000) [inline code/controllers/configuration.dm]: this actually should be nanotrasen.se/media; jukebox won't work without it

File: code/datums/datumvars.dm (hunk 4 above)
Comment (Markolie) [inline code/datums/datumvars.dm]: It shouldn't - it checks for a ckey.

File: config/example/dbconfig.txt (hunk 2 above)
Comment (tigercat2000) [inline config/example/dbconfig.txt]: eh, it's a moot point, anyone that can be bothered to setup SQL will probably rename it anyways, and if they can do that, they can edit this

File: code/controllers/configuration.dm (hunk 6 above)
Comment (Markolie) [inline code/controllers/configuration.dm]: Will ask Earth to update the URL - I'll add a proper config setting.

File: code/datums/datumvars.dm (hunk 4 above)
Comment (tigercat2000) [inline code/datums/datumvars.dm]: Really? Might have just been old code then. Or perhaps, ckeys aren't cleared until the client has been gone for a while.

File: code/controllers/configuration.dm (hunk 6 above)
Comment (tigercat2000) [inline code/controllers/configuration.dm]: my point was leaning more towards "if this isn't part of the default config people will probably not understand why the jukebox is not working"

Comment (Markolie) [inline code/controllers/configuration.dm]: Yea - I'll add it to the example file. That should be clear enough (right below the forum/wiki link).

**Comment by tigercat2000** [comment]
Merge conflicts, apparently.

**Comment by Spacemanspark** [comment]
B-b-but I liked Erro's poll!
:+1:

**Comment by Regen1** [comment]
I hate to be "that" guy, but

>Your pull requests should be atomic. Balance changes, bug fixes and new features should all be in separate pull requests.

**Comment by tigercat2000** [comment]
@Regen1 It would really be more trouble than it's worth to try and make atomic PR's for all of this.

**Comment by Regen1** [comment]
@tigercat2000 
It's a standard that should be followed.

**Comment by Alex6511** [comment]
Speaking from a professional standpoint it really doesn't matter.

**Comment by tigercat2000** [comment]
@Alex6511 okay I have to disagree with you there, 43 files is a huge amount for ss13
@Regen1 it probably should be, but it's more of "50 new PR's for each of these or just throw them all together since they need to get in around the same time"

**Comment by Alex6511** [comment]
@tigercat2000 I edited my post to be more accurate.

**Comment by Markolie** [comment]
Yea, I see little use in following that guideline (guideline, not rule) in cases like this.

**Comment by KasparoVy** [comment]
Does this also mean that during Highlander Vox/Plasmen won't get screwed as well?

**Comment by Markolie** [comment]
No, but I'll fix that. Thanks for the heads up.

**Comment by Markolie** [comment]
Okay, easy fix for Vox. I'm going to make it so Plasmamen spawn as humans in dodgeball/highlander though, as their space suit makes it hard to identify (impossible) what team they're on and gives them disproportionate protection.

**Comment by Robertha89** [comment]
well, it does not matter for dodgeball.

**Comment by KasparoVy** [comment]
Thanks a ton! Much appreciated fix

**Comment by Spacemanspark** [comment]
@Markolie Can't you make two new space suit sprites, blue and red, and do what the mining suit for plasmamen currently does (lowers their armor values.)?
Not saying what you have currently is bad, that's a pretty good alternative than being burned alive.
I'm just curious.

**Comment by Markolie** [comment]
If someone makes the sprites for them I don't mind implementing it.

**Comment by Spacemanspark** [comment]
@KasparoV
Or I might do it myself when I get home.

**Comment by kasparov** [comment]
I think you meant @KasparoVy

**Comment by Spacemanspark** [comment]
Oh.
Shit.
Sorry!

**Comment by KasparoVy** [comment]
@Spacemanspark I'll see about making some plasman sprites for this, no ETA though. I'll try and get them (among a few other things) going this weekend

**Comment by Spacemanspark** [comment]
Make it look like Master Chief. :^)
(And then we can do red vs blue references with the dodgeball suit sprites too! [Please don't kill me]).

**Pull request merged**